src/sentry/api/endpoints/relay_register.py

```diff
@@ -188,11 +188,14 @@ def post(self, request):
             relay.save()
 
         try:
-            relay_usage = RelayUsage.objects.get(relay_id=relay_id, version=version)
+            relay_usage = RelayUsage.objects.get(
+                relay_id=relay_id, version=version, public_key=public_key
+            )
         except RelayUsage.DoesNotExist:
             RelayUsage.objects.create(relay_id=relay_id, version=version)
         else:
             relay_usage.last_seen = timezone.now()
+            relay_usage.public_key = public_key
```

Comment:
Shouldn't this rather set the public_key during creation? Otherwise, you're creating usage entries without a key, but you're updating the key every time where it can never change.

```suggestion
            relay_usage = RelayUsage.objects.get(relay_id=relay_id, version=version)
        except RelayUsage.DoesNotExist:
            RelayUsage.objects.create(relay_id=relay_id, version=version, public_key=public_key)
        else:
            relay_usage.last_seen = timezone.now()
```

Reply:
right, the public_key argument from the getter should be in the create, sorry